app/src/main/java/org/gdg/frisbee/android/common/GdgNavDrawerActivity.java

```diff
@@ -178,20 +184,36 @@ private void onDrawerItemClick(final MenuItem item) {
                 }
                 break;
             case Const.DRAWER_HOME:
-                navigateTo(MainActivity.class, data);
+                if (!(this instanceof MainActivity)) {
+                    navigateTo(MainActivity.class, data);
+                }
                 break;
             case Const.DRAWER_GDE:
-                navigateTo(GdeActivity.class, data);
+                if (!(this instanceof GdeActivity)) {
+                    navigateTo(GdeActivity.class, data);
```

Comment:
It just occured to me that we can put these if statements into `navigateTo` function right? It would be more clear.